### ISSUE TITLE
Bug: [no-unnecessary-condition] `for(;true;)` should be allowed when `allowConstantLoopConditions: "only-allowed-literals"`

### ISSUE THREAD

**Issue by zanminkian** [typescript-eslint/typescript-eslint#11978]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.ts&code=PTAEGEHsCdoUwMYBdQMgEzgKAO4AsBLAGzlAAoloBXOASlAG8tRQRQA6TrAXyyzYAqhAM6gAhkWGRQwvJCpF0oAEak0sRCjSYsAMxjkSKAqAC8oAAwBuUJRo2CAakf0mLNp3Y8sQA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1tiacTJTIAhtEK0yHJgBNK%2BSpPRgA2uGw5E0aB2iQANKrVY12SEPjwOAdwDCk5PiEsAMhw7Fb02fKaoMkSfBiZhaWiFK0jPgaZqgG2AC%2BcQC6qonxQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
// Correct code
while (true) {
  // ...
}

// This also should be correct code
for (let i = 0; true; i++) {
  // ...
}
```

### ESLint Config

```javascript
{
  "rules": {
    "@typescript-eslint/no-unnecessary-condition": [
      "error",
      {
        "allowConstantLoopConditions": "only-allowed-literals"
      }
    ]
  }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    "strictNullChecks": true
  }
}
```

### Expected Result

No errors.

### Actual Result

There is an error on `for (let i = 0; true; i++)`

**Comment by undsoft** [comment]
Same for `do {} while (true)`
